core/domain/stats_jobs_one_off.py

```diff
@@ -127,3 +143,2369 @@ def reduce(key, stringified_values):
+                    assumptions[0:-1], operators),
+                AnswerMigrationJob._display_expression_helper(
+                    assumptions[-1], operators, 0))
+
+        target_string = AnswerMigrationJob._display_expression_helper(
+            results[0], operators, 0)
+
+        # The resulting answer is only 'Correct' if it is not a default answer,
+        # since if it were correct then it would have been matched to the
+        # 'Correct' rule type.
+        correct = rule_str != cls._DEFAULT_RULESPEC_STR
+        if correct:
+            return cls._normalize_raw_answer_object(objects.CheckedProof, {
+                'assumptions_string': assumptions_string,
+                'target_string': target_string,
+                'proof_string': answer_str,
+                'correct': True
+            }, answer_str)
+        else:
+            return cls._normalize_raw_answer_object(objects.CheckedProof, {
+                'assumptions_string': assumptions_string,
+                'target_string': target_string,
+                'proof_string': answer_str,
+                'correct': False,
+                'error_category': '',
+                'error_code': '',
+                'error_message': '',
+                'error_line_number': -1
+            }, answer_str)
+
+    @classmethod
+    def _cb_reconstitute_math_expression_input(
+            cls, state, answer_group_index, rule_spec, rule_str, answer_str):
+        if rule_str != cls._DEFAULT_RULESPEC_STR:
+            if rule_spec.rule_type != 'IsMathematicallyEquivalentTo':
+                return (
+                    None,
+                    'Cannot reconstitute MathExpressionInput object without an '
+                    'IsMathematicallyEquivalentTo rule.')
+        math_expression_dict = ast.literal_eval(answer_str)
+        if not isinstance(math_expression_dict, dict):
+            return (
+                None,
+                'Bad answer string in MathExpressionInput '
+                'IsMathematicallyEquivalentTo rule.')
+        return cls._normalize_raw_answer_object(
+            objects.MathExpression, math_expression_dict, answer_str)
+
+    @classmethod
+    def _cb_reconstitute_multiple_choice_input(
+            cls, state, answer_group_index, rule_spec, rule_str, answer_str):
+        # The Jinja representation for NonnegativeInt answer strings is:
+        #   {{ choices[answer|int] }}
+        interaction = state.interaction
+        if rule_str == cls._DEFAULT_RULESPEC_STR or (
+                rule_spec.rule_type == 'Equals'):
+            customization_args = interaction.customization_args
+            choices = customization_args['choices']['value']
+
+            # An answer group index is only available for non-default rules.
+            if rule_str != cls._DEFAULT_RULESPEC_STR:
+                outcome = interaction.answer_groups[answer_group_index].outcome
+                temporary_special_param = next(
```

Comment:
Not well documented, but also hard to explain. :) If you take a look at ``_force_rule_spec_subjects_to_answer`` you'll see that we add a special parameter for cases when the subject of the interaction handler is not an answer (this is mainly for cities and derivative explorations to properly retain their submitted answers). It's complicated, but it seems to work.